File: backend/push-service/src/test/java/com/example/pushservice/api/pub/PushPublicControllerTest.java
package com.example.pushservice.api.pub;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertNull;

import org.junit.jupiter.api.Test;

class PushPublicControllerTest {

    @Test
    void unquote_stripsJsonStringQuotes() {
        assertEquals("abc:DEF-123", PushPublicController.unquote("\"abc:DEF-123\""));
    }

    @Test
    void unquote_leavesPlainTokenAlone() {
        assertEquals("abc:DEF-123", PushPublicController.unquote("abc:DEF-123"));
    }

    @Test
    void unquote_trimsWhitespace() {
        assertEquals("abc", PushPublicController.unquote("  \"abc\"\n"));
    }

    @Test
    void unquote_null() {
        assertNull(PushPublicController.unquote(null));
    }
}

File: backend/push-service/src/test/kotlin/com/example/pushservice/api/pub/PushPublicControllerTest.kt
package com.example.pushservice.api.pub

import org.junit.jupiter.api.Test
import kotlin.test.assertEquals
import kotlin.test.assertNull

class PushPublicControllerTest {

    @Test
    fun unquote_stripsJsonStringQuotes() = assertEquals("abc:DEF-123", PushPublicController.unquote("\"abc:DEF-123\""))

    @Test
    fun unquote_leavesPlainTokenAlone() = assertEquals("abc:DEF-123", PushPublicController.unquote("abc:DEF-123"))

    @Test
    fun unquote_trimsWhitespace() = assertEquals("abc", PushPublicController.unquote("  \"abc\"\n"))

    @Test
    fun unquote_null() = assertNull(PushPublicController.unquote(null))
}
